src/NaCl.Core/Base/ChaChaIntrinsics/ChaCha256.cs

```diff
@@ -0,0 +1,174 @@
+﻿#if INTRINSICS
```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it